docs/certificate support proposal.md

```diff
@@ -0,0 +1,229 @@
+# Why? 
+
+## Client Certificates 
+Web apps and Web APIs are confidential client applications.
+
+They can prove their identity to Azure AD by 3 means:
+- client secrets
+- client certificates
+- client assertions
+
+Today, Microsoft.Identity.Web enables developers to provide client secrets.
+In addition to Client secret, we'd want Microsoft.Identity.Web to support client certificates. The constraints are the following:
+- enable several ways of getting the certificate. You'd provide a description on how to get the certificate.
+  - from the certificate store (Windows) and a thumbprint ("440A5BE6C4BE2FF02A0ADBED1AAA43D6CF12E269")
+  - from the certificate store (Windows) and a distinguished name ("CN=TestCert")
```

Comment:
These will possibly also work from Linux apparently ... https://devblogs.microsoft.com/aspnet/configuring-https-in-asp-net-core-across-different-platforms/